src/plugins/security_oss/server/routes/anonymous_access_capabilities.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { IRouter } from 'kibana/server';
+import { AnonymousAccessService } from '../plugin';
+
+interface Deps {
+  router: IRouter;
+  getAnonymousAccessService: () => AnonymousAccessService;
+}
+
+/**
+ * Defines route that returns capabilities of the anonymous service account.
+ */
+export function setupAnonymousAccessCapabilitiesRoute({ router, getAnonymousAccessService }: Deps) {
+  router.get(
+    { path: '/internal/security_oss/anonymous_access/capabilities', validate: false },
+    async (_context, request, response) => {
+      return response.ok({
+        body: await getAnonymousAccessService().getCapabilities(request),
```

Comment:
It looks like this route is always expecting the `anonymous access service` to be defined, but since it's only registered by x-pack security (as far as I can see), I think this will throw an [exception](https://github.com/elastic/kibana/blob/0fad3e6463274775138e25e2bf7258957746f617/src/plugins/security_oss/server/plugin.ts#L73) for the OSS distribution whenever its called.

Reply:
Yeah, it will, and user will get `500` error. Would you like it to have some default `200` behavior (then we'll have to have default capabilities in two places) or just more appropriate error code (e.g. `501 Not Implemented` or `403 Forbidden`)?

Reply:
I like `501 Not Implemented` for this case. I assume that this endpoint won't normally be called if anonymous access is disabled (and by extension, when called in the OSS distribution)? In other words, we'll have enough information client-side so that we know not to even attempt the call

Reply:
> I assume that this endpoint won't normally be called if anonymous access is disabled (and by extension, when called in the OSS distribution)?

Correct, there shouldn't be any case when Kibana would call it with disabled anonymous access.